packages/backend-data/src/convert_schema.ts

```diff
@@ -16,7 +21,22 @@ export const isModelSchema = (
   return (
     schema !== null &&
     typeof schema === 'object' &&
-    typeof schema.transform === 'function'
+    Object.hasOwn(schema, 'transform')
```

Comment:
These types are disunite and casting into any to refine with brackets made the package linter extra unhappy. I don't expect this to be the answer, happy to take input on how to best type guard these.